j5/backends/backend.py

```diff
@@ -32,6 +32,10 @@ def __new__(mcs, name, bases, namespace, **kwargs):  # type:ignore
         if cls.__name__ == "Backend":
             return cls
 
+        # Check if this is an abstract Backend.
+        if len(cls.__bases__) == 1 and cls.__base__.__name__ != "Backend":
```

Comment:
What is the aim of this bit? I don't really understand what its doing, and it looks like a bit of a hack. Don't most (if not all) concrete `Backend`s have more than one base class anyway?

Reply:
This identifies abstract Backends such as `RawUSBHardwareBackend`, which do not have a specific board associated with them and exempts them from the check.

Should no longer be necessary when #149 is done.

Reply:
-> Hardware PR